src/buffer.rs

```diff
@@ -243,6 +248,19 @@ impl Buffer {
         Ok(())
     }
 
+    /// Takes other buffer, measures its length and prints this buffer from the point where
+    /// the other stopped.
+    /// Used to implement autosuggestions.
+    pub fn print_rest<W>(&self, out: &mut W, other: &Buffer) -> io::Result<usize>
+        where W: Write
+    {
+        let matching_part = other.data.len();
+        let string: String = self.data.iter().skip(matching_part).cloned().collect();
+        out.write(string.as_bytes())?;
```

Comment:
I think it might be a bit strange to have a method this way - after all, there's no guarantee here that `other.data == self.data[matching_part..]`.

I think maybe `pub fn print_after<W>(&self, out: &mut W, after: usize) ...` would be better.